### DISCUSSION TITLE
Update existing image from byte array with raw image data

### DISCUSSION THREAD

**Discussion by ggolda** [SixLabors/ImageSharp#1682] [Q&A] [answered]
Is it possible to update/mutate Image instance with a pixel data from a byte array?
Similar to this method `Image.LoadPixelData<Bgr24>(pixels, width, height)` but without creating a new Image instance?

My use case is that I fetch raw byte array with pixels from OpenGL and need to draw them as an Image, that will be almost immediately discarded. As I know, you use a lot of memory allocation optimization and etc, but in my specific example here maybe it is possible just to update an existing image instead?

Thanks!

**Comment by JimBobSquarePants** [comment, marked as answer]
Yep. That's something a few people have asked for over the years.

[Image.WrapMemory](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Image.html#SixLabors_ImageSharp_Image_WrapMemory__1_System_Buffers_IMemoryOwner___0__System_Int32_System_Int32_)

  ```c#
  var memory = new Memory<Rgba32>(array);
  using Image<Rgba32> image = Image.WrapMemory(memory, width, height);
  ```

**Reply by antonfirsov** [reply]
@ggolda although based on your wording, you probably asked for a method like `existingImageInstance.CopyPixelDataFrom(somePixelSpanOrArray, w, h)`, `WrapMemory` might be actually better for you, depending on what you do.

You can also `WrapMemory` an unmanaged buffer. You can't apply operations that would alter the size, otherwise it works like a normal `Image` with the difference that the underlying buffer is not controlled by ImageSharp.

> [...] and need to draw them as an Image

What do you mean by this? Can you provide more details? Where does the output go? Are you sending it back to OpenGL?

**Reply by ggolda** [reply]
Thanks @JimBobSquarePants, that was exactly a method that I was looking for.
